src/components/ha-date-range-picker.ts

```diff
@@ -343,39 +294,64 @@ export class HaDateRangePicker extends LitElement {
   }
 
   private _handleNext(): void {
-    const dateRange = [
-      roundToNearestHours(this.endDate),
-      subMilliseconds(
-        roundToNearestHours(
-          addMilliseconds(
-            this.endDate,
-            Math.max(
-              3600000,
-              differenceInMilliseconds(this.endDate, this.startDate)
+    let dateRange: [Date, Date];
```

Comment:
Lets extract the next and prev functions from `hui-energy-period-selector` and make them helper functions we can use here too.

Reply:
Not sure if the logic from energy-period-selector should be used, as the new functions are more universal, which means they can deal with any-ranges and also shorter than one day. If selected a week the range is 7 days: so Week is not needed. Extra logic is only needed for month, as the timespans can be different. And the month-Logic can also shift between years without problem.

Reply:
Still they share the same logic right? So we can combine them into 1 function we use in both places?

Reply:
yes, took now a detailed look at it: you are right: we can combine them. Will upload a draft: hopefully that's the way it should be done ..